### PR TITLE
[server] simplify workspace classes

### PR DESCRIPTION
**Pull request by svenefftinge**
Based on & includes https://github.com/gitpod-io/gitpod/pull/15090

## Description
This PR reduces complexity about what workspace classes is used by
- removing user-level setting
- removing deprecation
- removing ws-class inheritence between ws sessions

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/15080
Related https://github.com/gitpod-io/ops/pull/7076

## How to test

Create a project and try the workspace class selection in its settings.

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [x] /werft with-payment
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@svenefftinge: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/config.ts [hunk 1]
```diff
@@ -307,7 +307,12 @@ export namespace ConfigFile {
             }
         }
 
-        WorkspaceClasses.validate(config.workspaceClasses);
+        if (config.workspaceClasses.filter((c) => c.isDefault).length !== 1) {
+            log.error(
+                "Exactly one default workspace class needs to be configured: " +
```
Comment (jankeromnes) [inline components/server/src/config.ts:312]: Nit: Could be clarified as "Exactly one workspace class should be configured as default: "

File: dev/preview/workflow/preview/deploy-gitpod.sh [hunk 2]
```diff
@@ -282,7 +282,6 @@ yq w -i "${INSTALLER_CONFIG_PATH}" experimental.webapp.workspaceClasses[1].categ
 yq w -i "${INSTALLER_CONFIG_PATH}" experimental.webapp.workspaceClasses[1].displayName "Small"
 yq w -i "${INSTALLER_CONFIG_PATH}" experimental.webapp.workspaceClasses[1].description "Small workspace class (20GB disk)"
 yq w -i "${INSTALLER_CONFIG_PATH}" experimental.webapp.workspaceClasses[1].powerups "2"
-yq w -i "${INSTALLER_CONFIG_PATH}" experimental.webapp.workspaceClasses[1].isDefault "false"
 yq w -i "${INSTALLER_CONFIG_PATH}" experimental.webapp.workspaceClasses[1].deprecated "false"
```
Comment (jankeromnes) [inline dev/preview/workflow/preview/deploy-gitpod.sh:285]: Whoops, did you mean to remove `.deprecated` instead of `.isDefault` here?

**Review by jankeromnes — APPROVED** [review]
Many thanks @svenefftinge! This second PR also looks good 👍 (modulo the rebase and previous comments to be addressed)

Also, holding due to one question: The UX generally works really well and is quite intuitive -- set class on project, get class on project workspaces 👍 not having a global user preference simplifies user mental model a lot. 💯

The only place where I'm a bit puzzled is workspaces based on prebuilds:
- When I start a workspace, it always gets the class currently defined on project-level (regardless of any prebuild class)
- This means that, when I switch from one class to another, various workspaces might get a different class to the prebuild they're based on
- I guess this is generally fine (very simple and predictable behavior), but it could lead to problems e.g. when the prebuild backup size is larger than the current workspace class allows, right?

Anyway, this is 99% fine by me, and my point is mostly food for thought, but I'm raising in case this wasn't intended or if we want to do anything about the error cases where you can't start a workspace due to a specified class that is not compatible with the prebuild's class.

/hold

**Comment by svenefftinge** [comment]
> but it could lead to problems e.g. when the prebuild backup size is larger than the current workspace class allows, right?

Yes, the workspace start would fail and should tell you that the class is incompatible. It currently doesn't do this, but this situation would be a rare corenercase, because the rebuild would need to have created a backup bigger than 20GB which is super rare.

**Review by vulkoingim — APPROVED** [review]

**Pull request merged**